### PR TITLE
feat: Added routeType type

### PR DESCRIPTION
**Pull request by johnlokerse**
## Description

Added a new user-defined type called `routeType`.

| Pipeline |
| - |
| [avm.res.network.route-table](https://github.com/johnlokerse/bicep-registry-modules/actions/workflows/avm.res.network.route-table.yml) |

### CODE REVIEW

File: avm/res/network/route-table/main.bicep [hunk 1]
```diff
@@ -132,3 +132,22 @@ type roleAssignmentType = {
   @description('Optional. The Resource Id of the delegated managed identity resource.')
   delegatedManagedIdentityResourceId: string?
 }[]?
+
+type routeType = {
+  @description('Name of the route.')
```
Comment (AlexanderSehr) [inline avm/res/network/route-table/main.bicep:137]:
```suggestion
  @description('Required. Name of the route.')
```

File: avm/res/network/route-table/main.bicep [hunk 2]
```diff
@@ -132,3 +132,22 @@ type roleAssignmentType = {
...
+  name: string
+
+  properties: {
+    @description('The type of Azure hop the packet should be sent to.')
```
Comment (AlexanderSehr) [inline avm/res/network/route-table/main.bicep:141]:
```suggestion
    @description('Required. The type of Azure hop the packet should be sent to.')
```

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Love it, please not that the requested changes will require a re-run of the `Set-AVMModule` script

File: avm/res/network/route-table/main.bicep (hunk 1 above)
Comment (johnlokerse) [reply avm/res/network/route-table/main.bicep:137]: Resolved!

**Comment by johnlokerse** [comment]
@AlexanderSehr Do you want me to test the change using a deployment?

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr Do you want me to test the change using a deployment?

Hey @johnlokerse, in general, for any module changes, that would be much appreciated, yes :)

**Comment by AlexanderSehr** [comment]
Small trick @johnlokerse  ;) 
image

Can be found in your pipeline via the 3 small `...` on the top right

**Comment by johnlokerse** [comment]
> Small trick @johnlokerse ;) image
> 
> Can be found in your pipeline via the 3 small `...` on the top right.
> 
> Adding it to your PR description.

That's great! Will do that the next time. Thanks for sharing :)

**Review by segraef — APPROVED** [review]
Thank you very much for your contribution @johnlokerse!

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**